astropy/units/core.py

```diff
@@ -726,7 +726,7 @@ def __eq__(self, other):
         try:
             other = Unit(other, parse_strict='silent')
         except (ValueError, UnitsError, TypeError):
-            return False
+            return NotImplemented
```

Comment:
This was not quite right before: the python standard is to return `NotImplemented` if one cannot compare to `other` sensibly - so that `other` gets a chance. If `other` then also returns `NotImplemented`, `False` is returned automatically. This might be useful for interaction with other unit packages (if they are aware of `astropy` units, and we're not aware of them).